lib/wiki_assignment_output.rb

```diff
@@ -39,19 +39,30 @@ def build_talk_page_update
     # Do not post templates to disambugation pages
     return nil if includes_disambiguation_template?(initial_page_content)
 
-    # We only want to add assignment tags to non-existant talk pages if the
+    # We only want to add assignment tags to non-existent talk pages if the
     # article page actually exists, and is not a disambiguation page.
     article_content = WikiApi.new(@wiki).get_page_content(@title)
     return nil if article_content.blank?
     return nil if includes_disambiguation_template?(article_content)
 
+    # Empty initial page content implies non-existent page.
+    talk_page_does_not_exist = initial_page_content.empty?
+    prevent_removal_of_assignments_from_non_existent_talk_pages(talk_page_does_not_exist) { return }
```

Comment:
This looks like it will work, but it it's a little confusing to me. It would be clearer, in my opinion, to follow the structure of the other guard statements that are already present: `return nil if [some condition]`.

I'm guessing that you're trying to work around complexity warnings from Rubocop here; if so, I suggest extracting lines 44-46 into a helper method; all that code is basically answering the same question: is it a real, existing article? I'd suggest something like `return nil unless is_normal_article?`

For the new check, I think a comment along with something like `return nil if @assignment.empty? && initial_page_content.empty?` will be easy enough to understand.

Reply:
Yep, exactly, I was trying to deal with Cyclomatic Complexity while trying to modify the least amount of existing code. I like your suggestions, addressed changes on 89a350151b55f6cf54f4ba8718edf6c5ecd0c145